extras/bindings/c/src/SolverInterfaceC.cpp

```diff
@@ -16,6 +16,8 @@ void precicec_createSolverInterface(
 {
   std::string stringAccessorName(participantName);
   std::string stringConfigFileName(configFileName);
+  if (interface != nullptr)
+    delete interface;
```

Comment:
We should better throw an error message here for the moment. It's currently never allowed to call that function twice, even with a different `participantName`. I could imagine that users expect a different behavior if they see that they can call this twice.

Reply:
Is this done via `PRECICE_ASSERT` or do you have any other way when throwing errors in the C interface?

Reply:
`PRECICE_ASSERT` here would be consistent with the rest of the class, yes. In principle, we could change all assertions here to errors, but that's the story of another PR. We have no perfect policy yet how to treat such problems in the bindings.